src/azul/service/async_manifest_service.py

```diff
@@ -85,13 +86,16 @@ def start_or_inspect_manifest_generation(self,
         else:
             token = self.decode_token(token)
         request_index = token.get('request_index', 0)
-        time_or_location = self._get_manifest_status(token['execution_id'], request_index)
-        if isinstance(time_or_location, int):
+        time_or_manifest = self._get_manifest_status(token['execution_id'], request_index)
+        if isinstance(time_or_manifest, int):
             request_index += 1
             token['request_index'] = request_index
-            return time_or_location, f'{self_url}?token={self.encode_token(token)}'
-        elif isinstance(time_or_location, str):
-            return 0, time_or_location
+            url = f'{self_url}?token={self.encode_token(token)}'
```

Comment:
should we use `furl` for URL composition?

Reply:
It might be worth noting that using `furl` the token value will now be url encoded, causing any `=` padding characters in the token to be replaced with `%3D`. This however should only be a good thing.

Reply:
AH i did not realize that the token encoding was effected by this change. If you feel that this can become problematic please reverse it.